### ISSUE TITLE
[Feature] new lint rule to require that each block must end with # or #. or a new line.

### ISSUE THREAD

**Issue by EdSaleh** [astral-sh/ruff#11698]
Hello,

Please create a new lint rule to require that each block must end with # or #. or a new line.
This would work as a curly braces blocks found in C style languages.
Also, this could also be used to reformat python code by code editor to fix any indentation issues.

Users can also use a different word symbol or use a new line for this purpose. (#., #, #/, newline, etc). The user has the option to select the keyword they would like to have for the project.

Example:

```python
import string
def atbash(sequence: str) -> str:
    """
    Example
    """
    output = ""
    for i in sequence:
        extract = ord(i)
        if 65 <= extract <= 90:
            output += chr(155 - extract)
        #.
        elif 97 <= extract <= 122:
            output += chr(219 - extract)
        #.
        else:
            output += i
        #.
    #.
    return output
#.
```

There are cases where strict block guarantees is required. Also, when copying python code sometimes spacing is not aligned correctly and having code with this style with editor support, editor would fix incorrect spacing easily.

```python
if(True):
print(True)
print(True)
#
if(True):
print(True)
print(True)
#
```

OR

```python
if(True):
print(True)
print(True)
#.
if(True):
print(True)
print(True)
#.
```
would be aligned correctly by the editor to:

```python
if(True):
 print(True)
 print(True)
#
if(True):
 print(True)
 print(True)
#
```
OR
```python
if(True):
 print(True)
 print(True)
#.
if(True):
 print(True)
 print(True)
#.
```
Users can also use new line instead of # or #. if they require and configure it to their requirements.

Thank you,

**Comment by AlexWaygood** [comment]
Hi, thanks for the suggestion! Unfortunately, I think very few of our users would want to switch this rule on, essentially for the same reasons explained to you by the pylint maintainers in https://github.com/pylint-dev/pylint/issues/9684. As such, I think this probably wouldn't be worth the maintenance burden for us, unfortunately. So we'll decline this one :-)

**Comment by EdSaleh** [comment]
I have also contributed to rust lint previously and my proposal was completed and merged.
https://github.com/rust-lang/rust-clippy/issues/2685

**Comment by zanieb** [comment]
Hi!

We use issues from other projects (pylint, flake8, black, etc.) to help inform our design decisions here as we want to have cohesion between various tools in the Python ecosystem.

However, regardless of the issue in pylint, I think that this rule would not be used by most of our users and is antithetical to general Python design principles. It is currently out of scope for what we'd want to include in Ruff. Maybe once we have support for plugins, it'd make sense there.

I have no problem with discussion continuing around this idea. Anyone is welcome to comment in this issue.
